lib/mysql_with_async/mysql_with_async.ml

```diff
@@ -0,0 +1,30 @@
+open Async
+open Core
+
+include Ppx_mysql_runtime.Make_context (struct
+  module IO = struct
+    type 'a t = 'a Deferred.t
+    let return = Deferred.return
+    let bind x f = Deferred.bind x ~f
+  end
+
+  module Prepared = struct
+    type dbh = Mysql.dbd
+    type stmt = Mysql.Prepared.stmt
+    type stmt_result = Mysql.Prepared.stmt_result
+    type error = [ `Mysql_exception of exn ]
+
+    let pool = Thread_pool.init ~name:"default" ~threads:8 ~create:Fn.id ~destroy:Fn.id
```

Comment:
We *may* want to create a database connection on `create` and close it on `destroy`. I am not sure